src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -83,10 +84,25 @@ public User findUserByEmail(String email) throws NotFoundException {
         return user;
     }
 
+    @Deprecated //instead use getUsersByUserRole(user, roleName)
     public Collection<User> describeUsers() {
         return userDAO.findUsers();
     }
 
+    public List<User> getUsersByUserRole(User user, String roleName) {
+        if (roleName.equalsIgnoreCase(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
+            if (Objects.nonNull(user.getInstitutionId())) {
+                return userDAO.findUsersByInstitution(user.getInstitutionId());
+            } else {
+                throw new NotFoundException("Signing Official (user: " + user.getDisplayName() + ") is not associated with an Institution.");
+            }
+        //there are only two cases here since we validate the roleName in the resource class, if it is not SO, then it is Admin
+        } else {
+            List<User> users = new ArrayList<>(userDAO.findUsers());
+            return users;
+        }
```

Comment:
It would be better to handle the Admin case explicitly here and not make this kind of assumption. If the role is admin, then return the dao call. If not, return an empty list.

Even better, you could add a switch on the role and have explicit cases for each role condition.